### PR TITLE
[New Model]: Certificate Signing Requests 1.0.0

### LINKED ISSUE #701: [New Model]: Certificate Signing Requests 1.0.0
## Model Description
Today, certificates for ECUs need to be provided by first and second tiers to the OEM. This is done by proprietary interfaces with the use of the CSR (Certificate Signing Request) approach. With this new aspect model, this process should be standardised based on the digital twins in accordance to the industry core. This will enable a data sovereign solution and reduce interfaces.

## Relevant Standards
CX-0125 Traceability Use Case or CX-0127 Industry Core: Part Instance (potentially R24.08) 

## Example Data
```json
{
    "certificateSigningRequestList": [
        {
            "certificateSigningRequest": "MIIC2TCCAToCAQAwgZwxEzARBgNVBAgTClByb2R1Y3Rpb24xDDAKBgNVBAcMA0FUTTESMBAGA1UEChMJQk1XIEdyb3VwMRcwFQYDVQQLEw5WZWhpY2xlUEtJLUVDVTEXMBUGA1UEAxMOSVBzZWMtaW50ZXJuYWwxMTAvBgNVBAUTKEVDVS1VSUQ6QzAwMDgyNDI3MUREQTYwMDAwMDAwMDAwMjdBQTg0MjQwdjAQBgcqhkjOPQIBBgUrgQQAIgNiAAShNm4y5Mdzu4b9xDW3TfBZr",
            "subjectLocality": "ATM--02",
            "subjectCommonName": "ID",
            "subjectSerialNumber": "SN12345678"
        },
        {
            "certificateSigningRequest": "MII3tbCCAToCAQAwgZwxEzARBgNVBAgTClByb2R1Y3Rpb24xDDAKBgNVBAcMA0FUTTESMBAGA1UEChMJQk1XIEdyb3VwMRcwFQYDVQQLEw5WZWhpY2xlUEtJLUVDVTEXMBUGA1UEAxMOSVBzZWMtaW50ZXJuYWwxMTAvBgNVBAUTKEVDVS1VSUQ6QzAwMDgyNDI3MUREQTYwMDAwMDAwMDAwMjdBQTg0MjQwdjAQBgcqhkjOPQIBBgUrgQQAIgNiAAShNm4y5Mdzu4b9xDW3TfBZr",
            "subjectLocality": "ATM--02",
            "subjectCommonName": "IPsec-Internal",
            "subjectSerialNumber": "SN12345678"
        }
    ]
}
```

## MS1 Criteria

- [x] The proposed aspect model does not exist already
- [x] The proposed aspect model does not extend an existing aspect but introduces completely new functionality
- [x] The proposal references a Catena-X use case
- [x] Relevant standards are mentioned/linked
- [x] A first draft of the data structure is provided in an example
- [x] A modeling steward has been assigned

### PR DESCRIPTION
**Pull request by johannsvarela**
## Description
Today, certificates for ECUs need to be provided by first and second tiers to the OEM. This is done by proprietary interfaces with the use of the CSR (Certificate Signing Request) approach. With this new aspect model, this process should be standardised based on the digital twins in accordance to the industry core. This will enable a data sovereign solution and reduce interfaces.

Closes #701 

## MS2 Criteria
(to be filled out by PR reviewer)
- [x] the model **validates** with the SAMM SDS SDK in the version specified in the Readme.md of this repository by the time of the MS2 check  (e.g., 'java -jar samm-cli.jar aspect \<path-to-aspect-model\> validate ). The  SAMM CLI is available [here](https://eclipse-esmf.github.io/esmf-developer-guide/tooling-guide/samm-cli.html) and in [GitHub](https://github.com/eclipse-esmf/esmf-sdk/releases/tag/v2.6.0)
- [x] use **Camel-Case** (e.g., "MyModelElement" or "TimeDifferenceGmtId", when in doubt follow https://google.github.io/styleguide/javaguide.html#s5.3-camel-case)
- [x] the identifiers for all model elements **start with a capital letter** except for properties
- [x] the identifier for **properties starts with a small letter**
- [x] all model elements **at least contain the fields "preferred name" and "description"** in English language. The description must be comprehensible. It is not required to write full sentences but style should be consistent over the whole model
- [x] Property and the referenced Characteristic should not have the same name
- [x] the versioning in the URN **follows semantic versioning**, where minor version bumps are backwards compatible and major version bumps are not backwards compatible. 
- [x] use **abbreviations only when necessary** and if these are sufficiently common
- [x] **avoid redundant prefixes in property names** (consider adding properties to an enclosing Entity or even adapt the namespace of the model elements, e.g., instead of having two properties `DismantlerId` and `DismantlerName` use an Entity `Dismantler` with the properties `name` and `id` or use a URN like `io.catenax.dismantler:0.0.1`)
- [x] fields `preferredName` and `description` are not the same
- [x] **`preferredName` should be human readable** and follow normal orthography (e.g., no camel case but normal word separation)
- [x] name of aspect is singular except if it only has one property which is a Collection, List or Set. In theses cases, the aspect name is plural.
- [x] units are referenced from the SAMM unit catalog whenever possible
- [x] **use constraints** to make known constraints from the use case explicit in the aspect model 
- [x] when relying on **external standards**, they are referenced through a **"see"** element
- [x] all properties with an [simple type](https://eclipse-esmf.github.io/samm-specification/2.0.0/datatypes.html) have an example value
- [x] metadata.json exists with status "release"
- [x] generated json schema validates against example json payload
- [x] file RELEASE_NOTES.md exists and contains entries for proposed model changes 
- [x] all contributors to this model are mentioned in copyright header of model file

## MS3 Criteria
(to be filled out by semantic modeling team before merge to main-branch)
- [x] All required reviewers have approved this PR (see reviewers section)
- [x] The new aspect (version) will be implemented by at least one data provider
- [x] The new aspect (version) will be consumed by at least one data consumer
- [x] There exists valid test data
- [x] In case of a new (incompatible) major version to an existing version, a migration strategy has been developed
- [x] The model has at least version '1.0.0'
- [x] If a previous model exists, model deprecation has been checked for previous model
- [x] The release date in the Release Note is set to the date of the MS3 approval

### CODE REVIEW

**Comment by johannsvarela** [comment]
modeling_team

FYI @BenediktMuellerIML

**Comment by johannsvarela** [comment]
@agg3fe @catroest Since we will urgently need this aspect model (for several internal implementations), we wanted to ask you whether it would be possible to review this PR by Monday 22.04 so that we can get MS2 & MS3 approval next week?

The new aspect model is about information that is needed to validate and certificate an ECU against the encrypted CSR data. For more details have a look at issue #701 or the descriptions within the aspect model. 

FYI @BenediktMuellerIML

File: io.catenax.certificate_signing_requests/1.0.0/CertificateSigningRequests.ttl [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+   samm:characteristic :CertificateSigningRequestCharacteristic;
+   samm:exampleValue "MIIC2TCCAToCAQAwgZwxEzARBgNVBAgTClByb2R1Y3Rpb24xDDAKBgNVBAcMA0FUTTESMBAGA1UEChMJQk1XIEdyb3VwMRcwFQYDVQQLEw5WZWhpY2xlUEtJLUVDVTEXMBUGA1UEAxMOSVBzZWMtaW50ZXJuYWwxMTAvBgNVBAUTKEVDVS1VSUQ6QzAwMDgyNDI3MUREQTYwMDAwMDAwMDAwMjdBQTg0MjQwdjAQBgcqhkjOPQIBBgUrgQQAIgNiAAShNm4y5Mdzu4b9xDW3TfBZr".
+
+:subjectSerialNumber a samm:Property;
+   samm:preferredName "Subject Serial Number"@en;
+   samm:description "The serial number of the assembly part generated by the manufacturer that is used to identifiy the ECU component in a unique way. The subject serial number is identical to the serial number (partInstanceId) specified in the aspect model SerialPart.\n\nThe value for the subjectSerialNumber is always the same for an ECU and its attached CSRs, but is required to additionally validate each individual CSR. "@en;
```
Comment (catroest) [inline io.catenax.certificate_signing_requests/1.0.0/CertificateSigningRequests.ttl:77]: typo: identify

File: io.catenax.certificate_signing_requests/1.0.0/CertificateSigningRequests.ttl [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+@prefix samm-e: <urn:samm:org.eclipse.esmf.samm:entity:2.1.0#>.
+@prefix unit: <urn:samm:org.eclipse.esmf.samm:unit:2.1.0#>.
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>.
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#>.
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#>.
+@prefix : <urn:samm:io.catenax.certificate_signing_requests:1.0.0#>.
+
+:CertificateSigningRequests a samm:Aspect;
+   samm:preferredName "Certificate Signing Requests"@en;
+   samm:description "The certificate signing requests model contains the essential information needed to request specific certificates for a unique ECU component via a Certificate Signing Request (CSR) and to validate and certificate the ECU against this data. "@en;
```
Comment (catroest) [inline io.catenax.certificate_signing_requests/1.0.0/CertificateSigningRequests.ttl:37]: delete space at the end

File: io.catenax.certificate_signing_requests/1.0.0/CertificateSigningRequests.ttl [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+   samm:properties ([ samm:property :certificateSigningRequestList; samm:optional true ]);
+   samm:operations ();
+   samm:events ().
+
+:certificateSigningRequestList a samm:Property;
+   samm:preferredName "Certificate Signing Request List"@en;
+   samm:description "Describes one or more Certificate Singing Requests (CSR). Each set entry contains additional information (e.g. subjectCommonName, subjectLocality, etc.) in readable form, which is otherwise part of the encrypted CSR. "@en;
```
Comment (catroest) [inline io.catenax.certificate_signing_requests/1.0.0/CertificateSigningRequests.ttl:44]: delete space at the end

File: io.catenax.certificate_signing_requests/1.0.0/CertificateSigningRequests.ttl [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+   samm:dataType :CertificateSigningRequestsEntity.
+
+:CertificateSigningRequestsEntity a samm:Entity;
+   samm:preferredName "Certificate Signing Requests Entity"@en;
+   samm:description "The Entity of a certificate signing requests to additionally structure the set."@en;
+   samm:properties (:subjectCommonName :subjectLocality :certificateSigningRequest :subjectSerialNumber).
+
+:subjectCommonName a samm:Property;
+   samm:preferredName "Subject Common Name\t"@en;
+   samm:description "Describes the common name on where the certificate is requested. "@en;
```
Comment (catroest) [inline io.catenax.certificate_signing_requests/1.0.0/CertificateSigningRequests.ttl:59]: delete space at the end

File: io.catenax.certificate_signing_requests/1.0.0/CertificateSigningRequests.ttl [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+   samm:characteristic :SubjectCommonNameCharacteristic;
+   samm:exampleValue "IPsec-Internal".
+
+:subjectLocality a samm:Property;
+   samm:preferredName "Subject Locality"@en;
+   samm:description "Provides information about the specific type of the ECU that will be certified. \n\nThe value for the subjectLocality is always the same for an ECU and its attached CSRs, but is required to additionally validate each individual CSR. "@en;
```
Comment (catroest) [inline io.catenax.certificate_signing_requests/1.0.0/CertificateSigningRequests.ttl:65]: delete space at the end

File: io.catenax.certificate_signing_requests/1.0.0/CertificateSigningRequests.ttl [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+   samm:characteristic :SubjectLocalityCharacteristic;
+   samm:exampleValue "ATM--02".
+
+:certificateSigningRequest a samm:Property;
+   samm:preferredName "Certificate Signing Request"@en;
+   samm:description "A CSR (Certificate Signing Request) is a 64-based PEM formatted and encrypted message to request and receive a digital certificate. The CSR validates the information required by a specific certification authority (CA) to issue the associated digital certificate.\n\nThe CSR consists of encrypted data, which can include, for example, basic information about the ECU and its manufacturer (common name, company, organizational unit, country, city/town, etc.), information about the public key attached to the certificate, as well as information on the key type and length. "@en;
```
Comment (catroest) [inline io.catenax.certificate_signing_requests/1.0.0/CertificateSigningRequests.ttl:71]: delete space at the end

File: io.catenax.certificate_signing_requests/1.0.0/CertificateSigningRequests.ttl (hunk 1 above)
Comment (catroest) [inline io.catenax.certificate_signing_requests/1.0.0/CertificateSigningRequests.ttl:77]: delete space at the end

File: io.catenax.certificate_signing_requests/1.0.0/CertificateSigningRequests.ttl [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+   samm:characteristic :CertificateSigningRequestCharacteristic;
+   samm:exampleValue "MIIC2TCCAToCAQAwgZwxEzARBgNVBAgTClByb2R1Y3Rpb24xDDAKBgNVBAcMA0FUTTESMBAGA1UEChMJQk1XIEdyb3VwMRcwFQYDVQQLEw5WZWhpY2xlUEtJLUVDVTEXMBUGA1UEAxMOSVBzZWMtaW50ZXJuYWwxMTAvBgNVBAUTKEVDVS1VSUQ6QzAwMDgyNDI3MUREQTYwMDAwMDAwMDAwMjdBQTg0MjQwdjAQBgcqhkjOPQIBBgUrgQQAIgNiAAShNm4y5Mdzu4b9xDW3TfBZr".
+
+:subjectSerialNumber a samm:Property;
```
Comment (catroest) [inline io.catenax.certificate_signing_requests/1.0.0/CertificateSigningRequests.ttl:75]: should this link or reference through a "see-link" to the serial part aspect?

**Review by catroest — CHANGES_REQUESTED** [review]
small changes

File: io.catenax.certificate_signing_requests/1.0.0/CertificateSigningRequests.ttl (hunk 1 above)
Comment (agg3fe) [inline io.catenax.certificate_signing_requests/1.0.0/CertificateSigningRequests.ttl:77]: typo `identifiy`

**Review by agg3fe — CHANGES_REQUESTED** [review]
please check all comments

**Comment by johannsvarela** [comment]
> please check all comments

@agg3fe @catroest  thanks for your rapid review! :-)  I've made changes as you commented. Please have a look at the adjustments.

**Review by agg3fe — APPROVED** [review]
LGTM

**Pull request merged**